frontend/src/index.html

```diff
@@ -143,58 +143,35 @@
     <div class="flex flex-1 mx-auto w-full max-w-screen-xl sm:overflow-y-hidden">
       <div id="filter-panel" class="fixed z-20 top-0 open:flex hidden flex-col w-64 max-w-full h-full text-sm bg-white md:static md:z-auto md:flex md:w-1/5 md:max-w-sm">
         <div class="flex-none p-2 text-center text-teal-800 font-bold bg-submarine-500 uppercase md:hidden">Filter</div>
-        <div class="flex-none p-2 pb-40 pt-0 overflow-x-hidden overflow-y-auto">
-          <!-- <div id="language-list"></div>
-          <div id="sound-quality-list"></div>
-          <div id="duration-menu"></div>
-          <div id="percentage-menu"></div>
-          <div id="category-list"></div>
-          <div id="year-list"></div>
-          <div id="location-list"></div> -->
-
-          <div>
-            <ul>
-              <li x-data="{selected:null}">
-                <button type="button" @click="selected !== 1 ? selected = 1 : selected = null">
-                  <div><span class="">LANGUAGES</span></div>
-                </button>
-                <div
-                  class="relative max-h-0 overflow-hidden transition-all duration-700"
-                  style=""
-                  x-ref="container1"
-                  x-bind:style="selected == 1 ? 'max-height: ' + $refs.container1.scrollHeight + 'px' : ''"
-                >
-                  <div>
-                    <div id="language-list"></div>
-                  </div>
-                </div>
-              </li>
-            </ul>
+        <div class="flex-grow p-2 pb-40 pt-0 overflow-x-hidden overflow-y-auto">
+          <div x-data="collapsible()">
+            <div x-on:click="collapseExpand()" class="ais-Panel-header">Languages</div>
+            <div id="language-list" x-show="open"></div>
           </div>
-
-          <div>
-            <ul>
-              <li x-data="{selected:null}">
-                <button type="button" @click="selected !== 1 ? selected = 1 : selected = null">
-                  <div><span>SOUND QUALITY</span></div>
-                </button>
-                <div
-                  class="relative max-h-0 overflow-hidden transition-all duration-700"
-                  style=""
-                  x-ref="container1"
-                  x-bind:style="selected == 1 ? 'max-height: ' + $refs.container1.scrollHeight + 'px' : ''"
-                >
-                  <div><div id="sound-quality-list"></div></div>
-                </div>
-              </li>
-            </ul>
+          <div x-data="collapsible()">
+            <div x-on:click="collapseExpand()" class="ais-Panel-header">Sound Quality</div>
+            <div id="sound-quality-list" x-show="open"></div>
```

Comment:
For the nested elements use `x-spread` to concentrate the logic in the JS file. See the audio player component for example. In this case it might be fine to keep the directives in HTML, because there are only two of them. But I'd suggest us to still use `x-spread` for you to learn it and for consistency.

Reply:
I am able to use x-spread but while calling the method **toggleCollapse()** , 
the following statement is not being executed":      **this.expanded = !this.expanded;**
 Any ideas?
---------------------------------
window.collapsible = () => {
  return {
    expanded: false,

    toggleCollapse() {
 this.expanded = !this.expanded;
    },
    toggle: {
      '@click': 'toggleCollapse',
    },
  };
};

Reply:
JS seems correct. Can you show me the corresponding HTML?

Reply:
Oh initially I missed the point that the method is called, but the inner statement is not executed. HTML is correct.

How do you know that the statement is not executed? Try installing the [Alpine.js devtools extension](https://chrome.google.com/webstore/detail/alpinejs-devtools/fopaemeedckajflibkpifppcankfmbhk) for Chrome and inspect the component's data.

By the way, now there is no `x-show` directive in your code, that is why changing the `expanded` property does not affect the component UI.

Reply:
I installed the same & here is what I can see.
Do we need any other directive?

![image](https://user-images.githubusercontent.com/15024800/130053506-efa14704-71f7-4af1-a586-5907f5b7d90b.png)

Reply:
Let's return to my question in the previous comment: How do you know that the statement is not executed?

Reply:
Actually, this.expanded = !this.expanded is executed and value changes but is not making an impact to do collapse and expand the section.

Reply:
Hari Bol @anantakrishna 

By adding the following, it works well:

class="overflow-hidden" x-bind:class="{'max-h-0': !expanded}"

![image](https://user-images.githubusercontent.com/15024800/130319720-c5cea3be-a4a1-42bb-bd58-53ecf5b838c3.png)

Reply:
I think `x-show` along with `transition` would be more straightforward in this case. Also, try to use `x-spread` in HTML and define all other directives in the JS file, as you did for the header (toggle).

By the way, it may be a good idea to rename `toggle` to `header` and `toggleCollapse` method to just `toggle`.

Reply:
After implementing this just push your new commits into the same branch and I will review the new version.

Reply:
done.
a better name would be **searchCriteriaHeader** or a little shorter **searchHeader** to be more specific.